Jf/fix kill npe for normal users

jayeff opened the pull request:
## Summary of the changes / Why this improves CrateDB

Trying to kill a non-existing job for a normal users leads to an exception: `NullPointerException[Cannot invoke "io.crate.execution.jobs.RootTask.userName()" because "ctx" is null]`

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

jayeff commented on server/src/test/java/io/crate/integrationtests/KillIntegrationTest.java:
```diff
@@ -48,7 +48,7 @@
 import static org.hamcrest.Matchers.is;
 
-public class KillIntegrationTest extends SQLTransportIntegrationTest {
+public class KillIntegrationTest extends BaseUsersIntegrationTest {
```
I switched to `BaseUsersIntegrationTest` to get access to `executeAsNormalUser`. Alternative could be to either extend `SQLTransportIntegrationTest` with such a method or add one to `KillIntegrationTest`. 

Wasn't sure what's the optimal solution for this. Would you prefer one of the alternatives?

mfussenegger replied:
I think it's fine this way :+1:

jayeff commented:
There are no tests for `KILL` for normal users. IMO missing are:
- trying to kill a job from another user => not allowed
- trying to kill your own job => happy path

I could push them onto this branch or create another PR for these tests.

mfussenegger reviewed (commented):
> There are no tests for KILL for normal users. IMO missing are:

I think both cases you mention are covered by this:

https://github.com/crate/crate/blob/c7d1c51ad0ac32d7060d9fcad20896acba8d05e8/server/src/test/java/io/crate/execution/jobs/TasksServiceTest.java#L230-L239

Changes look good to me. Could you squash it down to one commit?

mfussenegger commented:
> Do you think it's useful to have tests in both KillIntegrationTest and TasksServiceTest?

I think a unit test would be sufficient, but also okay to keep as is. Your choice.

We tend to prefer unit tests to test edge cases and use the full integration tests "only" to ensure that the components are wired correctly together or where unit testing becomes impractical due to the setup that's required.

Main reason is the execution speed. A integration test costs hundres of ms. A unit test a fraction of that.

jayeff commented:
> I think a unit test would be sufficient, but also okay to keep as is. Your choice.

I removed the integration test and rebased branch onto master.

I've the same opinion on unit and integration tests. Frankly I just completely missed `TasksServiceTest`. The failing integration test was my entry point into debugging and fixing this bug. Once it was fixed and the test passed I stopped there 🙂

mfussenegger reviewed (approved):
Nice, thank you for the fix :+1:
